distributed/scheduler.py

```diff
@@ -4693,24 +4700,26 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         if ts._dependencies or valid_workers is not None:
             ws = decide_worker(
                 ts,
-                self.workers.values(),
+                workers.values(),
                 valid_workers,
                 partial(self.worker_objective, ts),
             )
         else:
-            worker_pool = self.idle or self.workers.values()
-            n_workers = len(worker_pool)
+            worker_pool = self.idle or self.workers
+            worker_pool_dv = cast(dict, worker_pool)
+            n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
-                ws = min(worker_pool, key=operator.attrgetter("occupancy"))
+                ws = min(worker_pool_dv.values(), key=operator.attrgetter("occupancy"))
```

Comment:
This would shave off a bit more time as it is faster to call `.values()` on a `dict` (one C call) as opposed to a `SortedDict`. Though this would mean the `sort` doesn't preserve the `SortedDict`'s order. However it appears `test_correct_bad_time_estimate` expects the order to be preserved. So will go ahead and drop this, which fixes that issue.